app/src/main/java/com/shagalalab/constitution/ui/base/SearchableFragment.kt

```diff
@@ -0,0 +1,49 @@
+package com.shagalalab.constitution.ui.base
+
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.widget.SearchView
+import androidx.core.view.MenuItemCompat
+import androidx.fragment.app.Fragment
+import com.shagalalab.constitution.R
+
+open class SearchableFragment(resId: Int) : Fragment(resId) {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setHasOptionsMenu(true)
+        super.onCreate(savedInstanceState)
+    }
+
+    private lateinit var submitText: (text: String?) -> Unit
```

Comment:
Can you make `String?` just `String`? You might check before passing string to this function, and pass only if it is not null